public/controller/data_functions.php

```diff
@@ -1058,7 +1061,12 @@ function change_assignment_version($username, $semester,$course, $assignment_id,
         return;
     }
     $json = json_decode(removeTrailingCommas(file_get_contents($file)), true);
-    $json["active_assignment"] = $assignment_version;
+    $json["active_assignment"] = (int)$assignment_version;
+
+    date_default_timezone_set('America/New_York');
+    $version_change_time = date("Y-m-d H:i:s");
+    $version_change = array("version"=>(int)$assignment_version, "time"=>$version_change_time);
+    array_push($json["history"],$version_change);
```

Comment:
I'd rather see than the three above lines:
```
$json['history'][] = array("version" => $assignment_version, "time" => date("Y-m-d H:i:s");
```
I'm not sure if the cast is necessary or not, so you might need that, I'm not sure and would rather check without first.